redis/commands/cluster.py

```diff
@@ -594,3 +911,326 @@ def readwrite(self, target_nodes=None):
+
+    def cluster_addslots(
+        self, target_node: _TargetNodesT, *slots: EncodableT
+    ) -> Awaitable:
+        """
+        Assign new hash slots to receiving node. Sends to specified node.
+
+        :target_node: 'ClusterNode'
+            The node to execute the command on
+
+        For more information see https://redis.io/commands/cluster-addslots
+        """
+        return self.execute_command(
+            "CLUSTER ADDSLOTS", *slots, target_nodes=target_node
+        )
+
+    def cluster_addslotsrange(
+        self, target_node: _TargetNodesT, *slots: EncodableT
+    ) -> Awaitable:
+        """
+        Similar to the CLUSTER ADDSLOTS command.
+        The difference between the two commands is that ADDSLOTS takes a list of slots
+        to assign to the node, while ADDSLOTSRANGE takes a list of slot ranges
+        (specified by start and end slots) to assign to the node.
+
+        :target_node: 'ClusterNode'
+            The node to execute the command on
+
+        For more information see https://redis.io/commands/cluster-addslotsrange
+        """
+        return self.execute_command(
+            "CLUSTER ADDSLOTSRANGE", *slots, target_nodes=target_node
+        )
+
+    def cluster_countkeysinslot(self, slot_id: int) -> Awaitable:
+        """
+        Return the number of local keys in the specified hash slot
+        Send to node based on specified slot_id
+
+        For more information see https://redis.io/commands/cluster-countkeysinslot
+        """
+        return self.execute_command("CLUSTER COUNTKEYSINSLOT", slot_id)
+
+    def cluster_count_failure_report(self, node_id: str) -> Awaitable:
+        """
+        Return the number of failure reports active for a given node
+        Sends to a random node
+
+        For more information see https://redis.io/commands/cluster-count-failure-reports
+        """
+        return self.execute_command("CLUSTER COUNT-FAILURE-REPORTS", node_id)
+
+    async def cluster_delslots(self, *slots: EncodableT) -> List[bool]:
+        """
+        Set hash slots as unbound in the cluster.
+        It determines by it self what node the slot is in and sends it there
+
+        Returns a list of the results for each processed slot.
+
+        For more information see https://redis.io/commands/cluster-delslots
+        """
+        return await asyncio.gather(
+            *[self.execute_command("CLUSTER DELSLOTS", slot) for slot in slots]
+        )
+
+    def cluster_delslotsrange(self, *slots: EncodableT) -> Awaitable:
+        """
+        Similar to the CLUSTER DELSLOTS command.
+        The difference is that CLUSTER DELSLOTS takes a list of hash slots to remove
+        from the node, while CLUSTER DELSLOTSRANGE takes a list of slot ranges to remove
+        from the node.
+
+        For more information see https://redis.io/commands/cluster-delslotsrange
+        """
+        return self.execute_command("CLUSTER DELSLOTSRANGE", *slots)
+
+    def cluster_failover(
```

Comment:
this function should be `async def` and the execute commands should await. This way, if someone tries to await a function that raises an error, a second error won't pop up saying you tried to await an exception (I can't remember if this applies to Python, but that seems to be my logic for previous AsyncCommand mixing).

Reply:
Right now, in the core commands module, we just use the sync commands methods instead of a separate methods with async/await. I just followed the same here.